code/game/machinery/antigrav.dm

```diff
@@ -0,0 +1,124 @@
+/obj/machinery/antigrav
+	name = "antigrav generetor"
+	desc = "It temporarily disables gravity around."
+	icon_state = "GraviMobile"
+	density = 1
+	anchored = 0
+	use_power = 1
+	idle_power_usage = 10
+	active_power_usage = 10000
+	circuit = /obj/item/weapon/circuitboard/autolathe
+
+	var/on = FALSE
+	var/image/ball = null
+	var/area/area = null
+
+/obj/machinery/antigrav/proc/start()
+	area = get_area(src)
+	if(area.gravity_blocker && area.gravity_blocker != src)
+		src.visible_message(SPAN_NOTICE("\The [src] failed to start."))
+		return
+
+	area.gravity_blocker = src
+	area.has_gravity = FALSE
+
+	if(!on)
+		start_anim()
+
+	on = TRUE
+	use_power = 2
+	update_icon()
+
+/obj/machinery/antigrav/proc/stop()
+	if(area)
+		area.gravity_blocker = null
+		area.gravitychange(area.cached_gravity, area)
+		area = null
+
+	if(on)
+		stop_anim()
+
+	on = FALSE
+	use_power = 1
+	update_icon()
+
+/obj/machinery/antigrav/Process()
+	if(stat & NOPOWER || !on)
+		stop()
+		return
+
+
+/obj/machinery/antigrav/attack_hand(mob/user)
+	if(!on)
+		if(anchored)
+			start()
+		else
+			user << SPAN_WARNING("Fasten \the [src] to the floor first.")
+	else
+		stop()
+
+/obj/machinery/antigrav/attackby(var/obj/item/I, var/mob/user)
+	src.add_fingerprint(usr)
+	if (I.has_quality(QUALITY_BOLT_TURNING))
+		if (anchored)
+			if(!on)
+				user << SPAN_NOTICE("You begin to unfasten \the [src] from the floor...")
```

Comment:
Isn't there generic/base behaviour for this? We shouldn't need to keep putting wrench checks into every new machine

Reply:
This is most basic behaviour for this purpose, as far as i know.

Reply:
Sadly you are correct, i made an issue for this a while ago #1770 
We will just have to accept this for now